protocols/upnp/src/behaviour.rs

```diff
@@ -513,12 +513,7 @@ impl NetworkBehaviour for Behaviour {
                     self.mappings.renew(gateway, cx);
                     return Poll::Pending;
                 }
-                GatewayState::GatewayNotFound => {
-                    return Poll::Ready(ToSwarm::GenerateEvent(Event::GatewayNotFound));
-                }
-                GatewayState::NonRoutableGateway(_) => {
-                    return Poll::Ready(ToSwarm::GenerateEvent(Event::NonRoutableGateway));
-                }
```

Comment:
I think we should return them at least once! We can fix that with a boolean flag that we set after we've emitted them :)

Reply:
It should already be returned once:

https://github.com/libp2p/rust-libp2p/blob/c8b5f49ec2478338df2f7c323d572751f7b66b90/protocols/upnp/src/behaviour.rs#L389-L397

https://github.com/libp2p/rust-libp2p/blob/c8b5f49ec2478338df2f7c323d572751f7b66b90/protocols/upnp/src/behaviour.rs#L403-L404

Reply:
Ah good catch! Yeah this is fine then :)